### PR TITLE
Refactor score commands

### PR DESCRIPTION
**Pull request by kevinlul**
~Should replace #165~

## Checklist

- [x] I am following the [contributing guidelines]( https://github.com/AlphaKretin/emcee-tournament-bot/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: src/commands/forcescore.ts [hunk 1]
```diff
@@ -29,7 +28,21 @@ const command: CommandDefinition = {
 		if (scores.length < 2) {
 			throw new UserError("Must provide score in format `#-#` e.g. `2-1`.");
 		}
-		const response = await support.tournamentManager.submitScoreForce(id, player, scores[0], scores[1]);
+		// Check command syntax first to avoid a database round trip
+		const tournament = await support.database.authenticateHost(id, msg.author.id, TournamentStatus.IPR);
+		try {
+			// eslint-disable-next-line no-var
+			var { challongeId } = await support.database.getConfirmedPlayer(player, id);
```
Comment (AlphaKretin) [inline src/commands/forcescore.ts:35]: is var the only way to establish a variable inside a try block and use it after the catch? that's really frustrating. what about `try { let } finally {}`?

Comment (kevinlul) [reply src/commands/forcescore.ts:35]: `let` and `const` are block-scoped and introduced somewhere around ES5/2015. `var` is the classic JavaScript declaration and it's scoped to the function and hoisted to the top-level and beginning. This was the most natural way I could find for doing this as the assignment I want to perform also involves a destructure.

**Comment by kevinlul** [comment]
To complete the puzzle, in the else condition on pendingScore (i.e. not confirming the score but being the first to submit), we need to get the opponent's Discord snowflake (don't currently have this information) and then check appropriately.

**Comment by kevinlul** [comment]
There might be a case for the drop commands also cleaning the score cache, but the drop commands really don't have the key (match id) unless we add another Challonge call, and the score cache is cleared on round advance anyway.

**Comment by kevinlul** [comment]
Can you see what lines are and aren't covered in the two commands now? And also decide if checking the opponent is still in the server remains in scope (if not then this is ready for review).

forcescore network calls:
1. GET index matches via findClosedMatch
1. PUT match via submitScore
1. Potential getRESTUsername
1. Final Discord reply

score network calls:
1. GET index matches via findMatch
1. Success submission path: PUT match via submitScore
1. Discord direct message to opponent.
1. Potential getRESTUsername x2
1. Discord message to private channels.
1. The final Discord reply, regardless of branch

**Comment by AlphaKretin** [comment]
Missed lines in `forcescore`:
image
In `score`:
image
image
image

**Comment by AlphaKretin** [comment]
I think we can save handling leaves-while-offline for another time as long as drops are handled consistently now.

**Comment by kevinlul** [comment]
Okay. That's not in scope of this PR so I seem to have been distracted...

**Review by AlphaKretin — APPROVED** [review]
Assuming I correctly understand this PR's scope and effect (after some discussion), LGTM. One non-null warning to suppress.

**Comment by kevinlul** [comment]
Due to a browser fault the name of the squashed commit was still the old one, so I amended it.

**Pull request merged**